anchore_engine/utils.py

```diff
@@ -235,6 +235,42 @@ def run_command_list(
     return pipes.returncode, stdout_result, stderr_result
 
 
+def run_check(cmd, **kwargs):
```

Comment:
this isn't capturing syft stderr, which is all of the logger (debug/status) output.

Reply:
This function _is_ capturing `stderr`. Could you point to where you would expect to capture `stderr` that this function is missing?

This function logs the `stdout` always to DEBUG level, and `stderr` to ERROR level when there is a non-zero exit status. For production usage, I don't expect anyone to configure logging at DEBUG where (as you point out) it could potentially be in the MB range for some images.

Reply:
what I meant to point out is that the syft logging output (to stderr) is not being logged with the engine logger in all cases --here only when the return code != 0 (not what was intended with the issue).

But on second read, this is probably alright behavior --you will get all of the information you would want out of the logs for when something is going wrong.

(also, ack on the MB length and your debug comment... just as long as we know what will happen going in 👍 )

Reply:
this has been modified to:

* always log both stderr and stdout at DEBUG levels
* conditionally log stderr at ERROR level if the level is higher than DEBUG

Note that Engine's logging handling is backwards, so when I say _"higher than DEBUG"_ it is translated in code as _"lower than DEBUG"_

Reply:
how fun! thanks for the helpful comment